assets/app/scripts/controllers/create.js

```diff
@@ -47,33 +119,198 @@ angular.module('openshiftConsole')
       return false;
     }
 
-    function updateTemplates() {
-      // Check if we've loaded templates from both the openshift and project namespaces.
-      $scope.templatesLoaded = openshiftTemplates && projectTemplates;
-      $scope.instantApps = {};
-      $scope.otherTemplates = {};
-
-      // Categorize templates as instant apps or "other."
-      var categorizeTemplates = function(template) {
-        var uid = uidFilter(template);
-        if (isInstantApp(template)) {
-          $scope.instantApps[uid] = template;
+    function matchesKeyword(name, tags, keyword) {
```

Comment:
Yes, and I can go either way on this. It resulted in some unexpected matches. For instance, searching for "database" matched templates with descriptions like "An example application with no database." Maybe that's OK.

It's a little odd as well if there's only a match in the description because we don't show the description. It's not clear why it's a match.